handbook: New page for creating screenshots, GIFs, and videos (#952)
sourcegraph/about: handbook/marketing/adding_screenshots_screen_recording.md

# Adding a screenshot or screen recording (GIF and video)

> Note: This guide presumes macOS

## TL;DR

- Favor capturing only the part of the screen that’s required
- Keep GIFs 15 seconds or less
- GIFs are typically great for short "check this out" recordings, where videos are better at "how-to" or promotional style content

> - Keep GIF and images as small as possible with no retina sizes (just 1 x)

Reviewer: How do you know if it is retina sized? Is there a setting to look for or way to check for this?

Author: Added explanation "(ratio of image size to intended display size is 1:1)"

- If doing a full-screen recording, 1280x720 (720p) will produce easy to read results at default settings.
- If you’re producing visual content regularly, invest in paid apps (you can expense them too).

> - Chose CleanShot if just doing screenshots and GIFs.
> - Add ScreenFlow to your list if you plan on doing videos frequently, as you will appreciate editing and clip speed-up capabilities.

Reviewer: ```suggestion
   - Chose CleanShot if just doing screenshots and GIFs.
   - Add ScreenFlow to your list if you plan on doing videos frequently, as you will appreciate editing and clip speed-up capabilities.
```

- [Cleanshot](https://getcleanshot.com/)<br/>
Export screen recordings to GIF or video and nice features such as hiding desktop icons and notifications during recordings automatically. It does not have any editing or annotation tools for screen recordings though.

- [Quicktime (videos only)](https://support.apple.com/en-au/guide/quicktime-player/qtp97b08e666/mac)<br/>
Can capture a window, a selection of, or the entire screen, and in conjunction with the "Split Clip" functionality makes trimming the start and end easy. It’s a great tool that’s free on macOS.

- [ScreenFlow](https://www.telestream.net/screenflow/overview.htm)<br/>
Simply the best screen recording, editing, and production tool available. Designed primarily for video work (e.g. editing) but can export to GIF.

### Screen recording tips

Recording a great screencast is a large topic, but here are some essentials to keep in mind:

- Ensure it has an easy to demonstrate and clear purpose, e.g. show smart search field autocomplete, syntax highlighting, and error handling

> - Make it as short and simple as possible, but not so short, it’s hard to know what you’re demonstrating

Reviewer 2: ```suggestion
- Make it as short and simple as possible, but enough to show what you're demonstrating
```

- Only capture the part of the screen that is necessary. If the output is video, then aim for 16:9 ratio for YouTube
- If the entire screen needs to be captured, then your resolution should be 1280x720 (720p), as that will make on-screen elements easy to read
- Decide whether it should be standalone, or if it only makes sense when embedded with content
- Ideally, GIFs should less than 15-20 seconds in length, otherwise, it’s most likely better as a video
- Invest in learning a tool such as ScreenFlow if you need to annotate and/or edit your recordings

> - Does your GIF or video make sense without an accompanying explanation? If not, then a VoiceOver or text annotations can help a lot

Reviewer 2: ```suggestion
- Does your GIF or video make sense without an accompanying explanation? If not, then a voice-over or text annotations can help a lot
```

Like most things, practice makes perfect, but if you’re wanting to get some advice before taking the plunge, Ryan, Thorsten, and Quinn (sqs), have all created screencasts and can help you get started, so just reach out on Slack.

> 

Reviewer: A brief section on audio would be really helpful (I've just done it at the same time, but then can't do the nice speed-ups in different sections). 

```suggestion

### Adding audio or voice-overs

TODO

```

> ## Embedding GIFs

Reviewer: How do you add a nice intro page and ending page? How do we prefer to title these?

```suggestion

## Adding start/end screens to videos

TODO

## Embedding GIFs
```

</div>

Export your video in 16:9 ratio (should probably be 720p), then [upload your video to YouTube](https://studio.youtube.com/channel/UCOy2N25-AHqE43XupT9mwZQ):

1. Click on the **CREATE** button
2. Select your video to begin uploading
3. Add a title that has key words someone is likely to use in a search
4. Add a 1-2 sentence description of your video that contains a call-to-action for the user to learn more, activate the feature, or at least, install Sourcegraph. See other Sourcegraph videos that may have description content you can re-use
5. For thumbnail, either select the best option presented or generate your own by viewing the video fullscreen, taking a screenshot, then uploading it
6. Select "No, it's not made for kids"
7. Expand ‘MORE OPTIONS" to add the tags: "sourcegraph", "codesearch", "universalcodesearch"
8. Click NEXT, then NEXT again
9. Under Visibility, select **Unlisted** if the video only makes sense if embedded in content, or **Public**. If you think the video would benefit from approval first, leave set as **Unlisted** or **Draft**.
10. Click **SAVE**

> Then give the marketing team the heads0up that a new video has been published, letting them know what visibility state the video is currently in.

Reviewer: ```suggestion
Then give the marketing team the heads-up that a new video has been published, letting them know what visibility state the video is currently in.
```
